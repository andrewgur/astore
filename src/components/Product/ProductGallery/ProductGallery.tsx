import React, { FC } from 'react';
import ImageGallery, { ReactImageGalleryItem } from 'react-image-gallery';
import 'react-image-gallery/styles/css/image-gallery.css';
import './product.gallery.css';

type ProductGalleryProps = {
  images: ReactImageGalleryItem[];
};

export const ProductGallery: FC<ProductGalleryProps> = ({ images }) => {
  return (
    <div className="image__gallery">
      <ImageGallery
        items={images}
        showNav={false}
        showFullscreenButton={false}
        showPlayButton={false}
        additionalClass='gallery'
        />
    </div>
  )
 };