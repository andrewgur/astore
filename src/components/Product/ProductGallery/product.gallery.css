.image__gallery {
  width: 560px;
}
.image-gallery-thumbnails,
.image-gallery-thumbnails .image-gallery-thumbnails-container {
  display: flex;
  flex-wrap: wrap;
  width: 100%;
}
.image-gallery-thumbnail {
  width: 60px;
  height: 60px;
  overflow: hidden;
  margin-right: 1px;
  cursor: pointer;
  transition: opacity ease-in-out .3s;
  border: 2px solid transparent;
}
.image-gallery-thumbnail:hover {
  opacity: .7;
  border: 2px solid transparent;
}
.image-gallery-thumbnail.active, 
.image-gallery-thumbnail:focus  {
  border: 2px solid rgba(0,0,0,.15);
}
.image-gallery-content .image-gallery-slide .image-gallery-image {
  height: 745px;
  max-height: 745px;
}

@media (max-width: 1200px) { 
  .image__gallery {
    width: 440px;
  }
  .image-gallery-content .image-gallery-slide .image-gallery-image {
    max-height: 585px;
  }
}
@media (max-width: 960px) { 
  .image__gallery {
    margin: 0 auto;
    width: 350px;
  }
  .image-gallery-content .image-gallery-slide .image-gallery-image {
    max-height: 485px;
  }
}
@media (max-width: 768px) { 
  .image__gallery {
    width: 600px;
  }
  .image-gallery-content .image-gallery-slide .image-gallery-image {
    max-height: 800px;
  }
}


