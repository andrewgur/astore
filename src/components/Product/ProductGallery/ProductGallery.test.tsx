import React from 'react';
import { render } from '@testing-library/react';
import { ProductGallery } from './ProductGallery';

describe('ProductGallery Component', () => {
  const images = [
    {
      original: '1.jpg',
      thumbnail: '1.jpg'
    },
    {
      original: '2.jpg',
      thumbnail: '2.jpg'
    },
    {
      original: '3.jpg',
      thumbnail: '3.jpg'
    }
  ];

  it('should match snapshot', () => {
    const { container } = render(<ProductGallery images={images} />);
    expect(container).toMatchSnapshot();
  });
});