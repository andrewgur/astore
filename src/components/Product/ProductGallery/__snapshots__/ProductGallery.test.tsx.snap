// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGallery Component should match snapshot 1`] = `
<div>
  <div
    class="image__gallery"
  >
    <div
      aria-live="polite"
      class="image-gallery gallery"
    >
      <div
        class="image-gallery-content bottom"
      >
        <div
          class="image-gallery-slide-wrapper bottom"
        >
          <div
            class="image-gallery-swipe"
          >
            <div
              class="image-gallery-slides"
            >
              <div
                aria-label="Go to Slide 1"
                class="image-gallery-slide  center "
                role="button"
                style="display: inherit; transform: translate3d(0%, 0, 0);"
                tabindex="-1"
              >
                <img
                  alt=""
                  class="image-gallery-image"
                  height=""
                  loading="eager"
                  sizes=""
                  src="1.jpg"
                  srcset=""
                  title=""
                  width=""
                />
              </div>
              <div
                aria-label="Go to Slide 2"
                class="image-gallery-slide  right "
                role="button"
                style="display: inherit; transform: translate3d(100%, 0, 0);"
                tabindex="-1"
              >
                <img
                  alt=""
                  class="image-gallery-image"
                  height=""
                  loading="eager"
                  sizes=""
                  src="2.jpg"
                  srcset=""
                  title=""
                  width=""
                />
              </div>
              <div
                aria-label="Go to Slide 3"
                class="image-gallery-slide  left "
                role="button"
                style="display: inherit; transform: translate3d(-100%, 0, 0);"
                tabindex="-1"
              >
                <img
                  alt=""
                  class="image-gallery-image"
                  height=""
                  loading="eager"
                  sizes=""
                  src="3.jpg"
                  srcset=""
                  title=""
                  width=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="image-gallery-thumbnails-wrapper bottom thumbnails-swipe-horizontal"
        >
          <div
            class="image-gallery-thumbnails"
          >
            <nav
              aria-label="Thumbnail Navigation"
              class="image-gallery-thumbnails-container"
              style="transform: translate3d(0px, 0, 0); transition: all 450ms ease-out;"
            >
              <button
                aria-label="Go to Slide 1"
                aria-pressed="true"
                class="image-gallery-thumbnail active"
                tabindex="0"
                type="button"
              >
                <span
                  class="image-gallery-thumbnail-inner"
                >
                  <img
                    class="image-gallery-thumbnail-image"
                    src="1.jpg"
                  />
                </span>
              </button>
              <button
                aria-label="Go to Slide 2"
                aria-pressed="false"
                class="image-gallery-thumbnail"
                tabindex="0"
                type="button"
              >
                <span
                  class="image-gallery-thumbnail-inner"
                >
                  <img
                    class="image-gallery-thumbnail-image"
                    src="2.jpg"
                  />
                </span>
              </button>
              <button
                aria-label="Go to Slide 3"
                aria-pressed="false"
                class="image-gallery-thumbnail"
                tabindex="0"
                type="button"
              >
                <span
                  class="image-gallery-thumbnail-inner"
                >
                  <img
                    class="image-gallery-thumbnail-image"
                    src="3.jpg"
                  />
                </span>
              </button>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
